src/Messaging/MessageDataAssertion.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/*
+ * This file is part of the prooph/common.
+ * (c) 2014-2015 prooph software GmbH <contact@prooph.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Date: 8/25/15 - 3:27 PM
+ */
+namespace Prooph\Common\Messaging;
+
+use Assert\Assertion;
+
+/**
+ * Class MessageDataAssertion
+ *
+ * @package Prooph\Common\Messaging
+ * @author Alexander Miertsch <kontakt@codeliner.ws>
+ */
+final class MessageDataAssertion
+{
+    /**
+     * @param mixed $messageData
+     */
+    public static function assert($messageData)
+    {
+        Assertion::isArray($messageData, 'MessageData must be an array');
```

Comment:
You can typehint for array directly.

    public static function assert(array $messageData)